### DISCUSSION TITLE
Is is possible to continuous redraw?

### DISCUSSION THREAD

**Discussion by btraquair** [ArthurSonzogni/FTXUI#555] [Q&A] [answered]
I see that typically the screen is redrawn after an event. Is it possible to have the screen redrawn based on time if no event occurs? 

What I am trying to do is implement a clock and eventually add a stop watch. So I would like to have the time update regardless of what the user is doing. Perhaps this is something obvious that I have missed.

**Comment by ArthurSonzogni** [comment, marked as answer]
Hello @btraquair!
There is an example on the wiki: https://github.com/ArthurSonzogni/FTXUI/wiki/Screen#force-redraw

It uses a thread to periodically post event/task/animation:
```cpp
screen_interactive.PostEvent(Event::Custom)
```
or
```cpp
screen_interactive.RequestAnimationFrame();
```
----


If you don't want to add a thread, you can use the `Loop` to interleave ftxui loop with it:

```cpp
  Loop loop(&screen, component);
  int interation = 0;
  while (!loop.HasQuitted()) {
    loop.RunOnce();
    std::this_thread::sleep_for(std::chrono::milliseconds(10));

    interaction++;
    if (iteration % 10 == 0)
      screen.RequestAnimationFrame();
  }

```

Does it work?

**Reply by btraquair** [reply]
Thank you. yes, this is precisely what I was searching for.
